tests/unit/storage_v3_gc.cpp

```diff
@@ -0,0 +1,224 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "storage/v3/key_store.hpp"
+#include "storage/v3/property_value.hpp"
+#include "storage/v3/storage.hpp"
+
+using testing::UnorderedElementsAre;
+
+namespace memgraph::storage::v3::tests {
+
+// class StorageV3GcTest : public ::testing::Test {
+//  protected:
+//   void SetUp() override {
+//     ASSERT_TRUE(storage.CreateSchema(label, {SchemaProperty{property, common::SchemaType::INT}}));
+//   }
+
+//   Storage storage{Config{.gc = {.type = Config::Gc::Type::PERIODIC, .interval = std::chrono::milliseconds(100)}}};
+//   const LabelId label{storage.NameToLabel("label")};
+//   const PropertyId property{storage.NameToProperty("property")};
+// };
+VertexAccessor CreateVertexAndValidate(Storage::Accessor &acc, LabelId primary_label,
+                                       const std::vector<LabelId> &labels,
+                                       const std::vector<std::pair<PropertyId, PropertyValue>> &properties) {
+  auto vtx = acc.CreateVertexAndValidate(primary_label, labels, properties);
+  EXPECT_TRUE(vtx.HasValue());
+  return *vtx;
+}
+
+// TODO: The point of these is not to test GC fully, these are just simple
+// sanity checks. These will be superseded by a more sophisticated stress test
+// which will verify that GC is working properly in a multithreaded environment.
```

Comment:
I am not sure what do you mean `GC is working properly in a multithreaded environment`, but as we discussed the GC has to be streamlined into the single threaded access to the shards, as we don't want to deal with accesses from multiple threads simultaneously.

Reply:
That's not me these comments were left from the file I've copied, but I will remove this file, this is just to prove that we did not break anything in this PR

Reply:
As I disable the background thread for GC in my PR, I think this file is unnecessary to add for this short period.